Dockerfile-go-deps

```diff
@@ -16,3 +18,79 @@ RUN bin/dep version
 COPY Gopkg.toml Gopkg.lock ./
 RUN bin/dep ensure -vendor-only -v
 
+# Move the dependencies to where `go install` can see them.
+ENV GOPATH=/go
+RUN mv vendor/* $GOPATH/src/
```

Comment:
would `go install -v ./vendor/...` work here?

Reply:
> would go install -v ./vendor/... work here?

I tried that before. Unfortunately that tries to install too many (sub-)packages, some of which are apparently downloaded by `dep ensure` but aren't actually needed in the build, and some of those don't have their own dependencies installed (since they aren't needed).

For example:
```
vendor/github.com/spf13/cobra/cobra/cmd/licenses.go:22:2: cannot find package "github.com/spf13/viper" in any of:
        /go/src/github.com/runconduit/conduit/vendor/github.com/spf13/viper (vendor tree)
        /usr/local/go/src/github.com/spf13/viper (from $GOROOT)
        /go/src/github.com/spf13/viper (from $GOPATH)
```